src/components/Divider.tsx

```diff
@@ -1,70 +1,172 @@
 import { SystemStyleObject } from '../../styled-system/types'
 import Path from '../@types/Path'
+import State from '../@types/State'
+import ThoughtId from '../@types/ThoughtId'
 import { setCursorActionCreator as setCursor } from '../actions/setCursor'
-import { DIVIDER_MIN_WIDTH, DIVIDER_PLUS_PX } from '../constants'
+import { DIVIDER_MIN_WIDTH } from '../constants'
 import attributeEquals from '../selectors/attributeEquals'
+import { getAllChildrenAsThoughts } from '../selectors/getChildren'
 import rootedParentOf from '../selectors/rootedParentOf'
+import editingValueUntrimmedStore from '../stores/editingValueUntrimmed'
 import fastClick from '../util/fastClick'
 import head from '../util/head'
+import isDivider from '../util/isDivider'
+
+/** Custom hook to fetch divider-related data from the state. */
+const useDividerData = (path: Path) => {
+  return useSelector(
+    (state: State) => {
+      const parentPath = rootedParentOf(state, path)
+      const parentId = head(parentPath)
+      const grandParentPath = parentId ? rootedParentOf(state, parentPath) : []
+      const grandParentId = head(grandParentPath)
+      const children = parentId ? getAllChildrenAsThoughts(state, parentId) : []
+      const childrenWithoutDividers = children.filter(child => !isDivider(child.value))
+      const isOnlyChild = childrenWithoutDividers.length === 0
+      const isTableView =
+        (parentId && attributeEquals(state, parentId, '=view', 'Table')) ||
+        (grandParentId && attributeEquals(state, grandParentId, '=view', 'Table'))
+
+      let thoughtsAtSameDepth: { id: ThoughtId; value: string }[] = []
+
+      if (isTableView && isOnlyChild && grandParentId) {
+        const parentSiblings = getAllChildrenAsThoughts(state, grandParentId).filter(child => !isDivider(child.value))
+        thoughtsAtSameDepth = parentSiblings.flatMap(parent => {
+          const childrenOfParent = parent.id ? getAllChildrenAsThoughts(state, parent.id) : []
+          return childrenOfParent.filter(child => !isDivider(child.value))
+        })
+      }
+
+      return {
+        isOnlyChild,
+        isTableView,
+        children,
+        thoughtsAtSameDepth,
+      }
+    },
+    // Ensure the selector runs only when values change, excluding ref changes to avoid redundant re-renders.
+    (prev, next) => JSON.stringify(prev) === JSON.stringify(next),
+  )
+}
+
+let canvas: HTMLCanvasElement | null = null
+let context: CanvasRenderingContext2D | null = null
+
+/**
+ * Initializes the canvas and context for text width calculation.
+ */
+const initializeCanvas = () => {
+  if (!canvas) {
+    canvas = document.createElement('canvas')
+    context = canvas.getContext('2d')
+  }
+}
+
+/**
+ * Calculates the width of a single text string, considering trailing spaces.
+ */
+const calculateTextWidth = (text: string, font: string): number => {
+  initializeCanvas()
+  if (!context) return 0
+  context.font = font
+  return Math.ceil(context.measureText(text).width)
```

Comment:
I believe this is the cause of `6. Divider continues to grow after text wraps`. Here, we are looking at the width of the entire thought text without considering line wrapping. We would need to either account for text that wraps, or set a maximum width after calculating the full text width.